tests/test_middleware_interface.py

```diff
@@ -20,19 +20,64 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 import os.path
-import tempfile
 import unittest
+import unittest.mock
+import shutil
+
+import astropy.units as u
+
+import astro_metadata_translator
+from lsst.daf.butler import DataCoordinate
+from lsst.obs.base.formatters.fitsExposure import FitsImageFormatter
+from lsst.obs.base.ingest import RawFileDatasetInfo, RawFileData
+import lsst.resources
 
 from activator.middleware_interface import MiddlewareInterface
 from activator.visit import Visit
 
 
+def fake_file_data(filename, dimensions, instrument):
+    """Return file data for a mock file to be ingested.
+
+    Parameters
+    ----------
+    filename : `str`
+        Full path to the file to mock. Can be a non-existant file.
+    dimensions : `lsst.daf.butler.DimensionsUniverse`
+        The full set of dimensions for this butler.
+    instrument : `lsst.obs.base.Instrument`
+        The instrument the file is supposed to be from.
+
+    Returns
+    -------
+    data_id, file_data, : `DataCoordinate`, `RawFileData`
+        The id and descriptor for the mock file.
+    """
+    data_id = DataCoordinate.standardize({"exposure": 1, "detector": 1, "instrument": instrument.getName()},
+                                         universe=dimensions)
+    obs_info = astro_metadata_translator.makeObservationInfo(instrument=instrument.getName(),
+                                                             exposure_id=1,
+                                                             observation_id="1",
+                                                             physical_filter="r",
+                                                             exposure_time=30.0*u.second)
+    dataset_info = RawFileDatasetInfo(data_id, obs_info)
+    file_data = RawFileData([dataset_info],
+                            lsst.resources.ResourcePath(filename),
+                            FitsImageFormatter,
+                            instrument)
+    return data_id, file_data
+
+
 class MiddlewareInterfaceTest(unittest.TestCase):
     def setUp(self):
-        input_dir = os.path.join(os.path.abspath(os.path.dirname(__file__)), "data")
-        instrument = "TestCam"
-        self.output_dir = tempfile.TemporaryDirectory()
-        self.interface = MiddlewareInterface(input_dir, self.output_dir, instrument)
+        data_dir = os.path.join(os.path.abspath(os.path.dirname(__file__)), "data/")
+        input_repo = os.path.join(data_dir, "input_repo")
+        instrument = "lsst.obs.lsst.LsstCam"
+        self.input_data = os.path.join(data_dir, "input_data")
+        # Use the current test name in the local butler path.
+        suffix = self.id().split('.')[-1]
```

Comment:
That gives the name of the test method (e.g. `test_init`). Otherwise, you get all the file and module name, which aren't helpful for my purpose here ("use a separate local butler instance for each test method"). I've expanded the comment.